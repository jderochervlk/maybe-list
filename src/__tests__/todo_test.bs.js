// Generated by ReScript, PLEASE EDIT WITH CARE
'use strict';

var Jest = require("@glennsl/rescript-jest/src/jest.bs.js");
var Curry = require("rescript/lib/js/curry.js");
var React = require("react");
var TodoItem = require("../TodoItem.bs.js");
var ReactTestingLibrary = require("../external/ReactTestingLibrary.bs.js");

var item = {
  text: "one",
  completed: false,
  createdAt: 1.0,
  id: 1
};

Jest.test("onClick is called when clicking", (function (param) {
        var result = ReactTestingLibrary.render(undefined, undefined, undefined, undefined, undefined, React.createElement(TodoItem.make, {
                  todo: item,
                  onClick: (function (param) {
                      
                    })
                }));
        ReactTestingLibrary.act(function (param) {
              Curry._2(ReactTestingLibrary.FireEvent.click, undefined, ReactTestingLibrary.getByRole({
                        NAME: "Str",
                        VAL: "checkbox"
                      }, undefined, result));
            });
        return Jest.Expect.toMatchSnapshot(Jest.Expect.expect(result.container));
      }));

exports.item = item;
/*  Not a pure module */
