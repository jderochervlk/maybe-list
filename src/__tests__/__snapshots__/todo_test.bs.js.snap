// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onClick is called when clicking 1`] = `
<div>
  <li>
    one
    <input
      class="completed"
      type="checkbox"
    />
  </li>
</div>
`;
